src/session-pool.js

```diff
@@ -785,8 +790,8 @@ SessionPool.prototype.pingSession_ = function(session) {
         });
     })
     .catch(function() {
-      var index = self.borrowed_.indexOf(session);
-      self.borrowed_.splice(index, 1);
+      // session is borrowed but not released, it will cause a session leak.
+      self.release(session);
```

Comment:
I think we'll want to verify that the error thrown is not a 404, in same cases we do want to discard the session.

Reply:
So, do we discard the session and call close on the session or just discard it?

Reply:
If its a 404, I imagine its safe to simply discard it. @vkedia any input here?

Reply:
yeah it is safe to discard it.